### PR TITLE
introduce randomness into edgeConnect name and hostPattern during e2e test run

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

To avoid race condition, when we run e2e tests against same tenant, but different clusters:

```

=== RUN   TestStandard
    default.go:52: istio enabled: false
=== RUN   TestStandard/edgeconnect-install
=== RUN   TestStandard/edgeconnect-install/create_EC_configuration_on_the_tenant
=== RUN   TestStandard/edgeconnect-install/create_client_secret
=== RUN   TestStandard/edgeconnect-install/'test-edgeconnect-normal-03567'_edgeconnect_created
=== RUN   TestStandard/edgeconnect-install/'test-edgeconnect-normal-03567'_edgeconnect_phase_changes_to_'Running'
=== RUN   TestStandard/edgeconnect-install/check_EC_configuration_on_the_tenant
=== RUN   TestStandard/edgeconnect-install/delete_EdgeConnect_CR
=== RUN   TestStandard/edgeconnect-install/check_if_EC_configuration_is_deleted_on_the_tenant
=== RUN   TestStandard/edgeconnect-install-provisioner
=== RUN   TestStandard/edgeconnect-install-provisioner/create_edgeconnect_client_secret
=== RUN   TestStandard/edgeconnect-install-provisioner/'test-edgeconnect-provisioner-a262d'_edgeconnect_created
=== RUN   TestStandard/edgeconnect-install-provisioner/'test-edgeconnect-provisioner-a262d'_edgeconnect_phase_changes_to_'Running'
=== RUN   TestStandard/edgeconnect-install-provisioner/get_tenant_config
=== RUN   TestStandard/edgeconnect-install-provisioner/get_EC_status
=== RUN   TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_exists_on_the_tenant
=== RUN   TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern
=== RUN   TestStandard/edgeconnect-install-provisioner/update_hostPatterns
=== RUN   TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern2
=== RUN   TestStandard/edgeconnect-install-provisioner/delete_EC_custom_resource
=== RUN   TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_is_deleted_on_the_tenant
=== RUN   TestStandard/edgeconnect-install-k8s-automation
=== RUN   TestStandard/edgeconnect-install-k8s-automation/create_ServiceAccount
=== RUN   TestStandard/edgeconnect-install-k8s-automation/create_edgeconnect_client_secret
=== RUN   TestStandard/edgeconnect-install-k8s-automation/'test-edgeconnect-automation-a7fc7'_edgeconnect_created
=== RUN   TestStandard/edgeconnect-install-k8s-automation/'test-edgeconnect-automation-a7fc7'_edgeconnect_phase_changes_to_'Running'
^F2=== RUN   TestStandard/edgeconnect-install-k8s-automation/get_tenant_config
=== RUN   TestStandard/edgeconnect-install-k8s-automation/get_EC_status
=== RUN   TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_exists_on_the_tenant
=== RUN   TestStandard/edgeconnect-install-k8s-automation/check_hostPatterns_-_k8s_automation
=== RUN   TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_exists_on_the_tenant
=== RUN   TestStandard/edgeconnect-install-k8s-automation/delete_EC_custom_resource
=== RUN   TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_is_deleted_on_the_tenant
=== RUN   TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_is_deleted_on_the_tenant
--- PASS: TestStandard (70.69s)
    --- PASS: TestStandard/edgeconnect-install (29.35s)
        --- PASS: TestStandard/edgeconnect-install/create_EC_configuration_on_the_tenant (1.38s)
        --- PASS: TestStandard/edgeconnect-install/create_client_secret (0.38s)
        --- PASS: TestStandard/edgeconnect-install/'test-edgeconnect-normal-03567'_edgeconnect_created (0.42s)
        --- PASS: TestStandard/edgeconnect-install/'test-edgeconnect-normal-03567'_edgeconnect_phase_changes_to_'Running' (20.19s)
        --- PASS: TestStandard/edgeconnect-install/check_EC_configuration_on_the_tenant (0.51s)
        --- PASS: TestStandard/edgeconnect-install/delete_EdgeConnect_CR (5.40s)
        --- PASS: TestStandard/edgeconnect-install/check_if_EC_configuration_is_deleted_on_the_tenant (0.40s)
    --- PASS: TestStandard/edgeconnect-install-provisioner (18.60s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/create_edgeconnect_client_secret (0.27s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/'test-edgeconnect-provisioner-a262d'_edgeconnect_created (0.23s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/'test-edgeconnect-provisioner-a262d'_edgeconnect_phase_changes_to_'Running' (5.22s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/get_tenant_config (0.33s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/get_EC_status (0.17s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_exists_on_the_tenant (0.36s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern (0.35s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/update_hostPatterns (5.38s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern2 (0.37s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/delete_EC_custom_resource (5.37s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_is_deleted_on_the_tenant (0.33s)
    --- PASS: TestStandard/edgeconnect-install-k8s-automation (19.98s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/create_ServiceAccount (0.20s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/create_edgeconnect_client_secret (0.18s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/'test-edgeconnect-automation-a7fc7'_edgeconnect_created (0.18s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/'test-edgeconnect-automation-a7fc7'_edgeconnect_phase_changes_to_'Running' (10.17s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/get_tenant_config (0.33s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/get_EC_status (0.18s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_exists_on_the_tenant (0.37s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_hostPatterns_-_k8s_automation (0.92s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_exists_on_the_tenant (0.66s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/delete_EC_custom_resource (5.38s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_is_deleted_on_the_tenant (0.31s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_is_deleted_on_the_tenant (0.70s)
PASS
out: helm uninstall dynatrace-operator \
                        --namespace dynatrace
release "dynatrace-operator" uninstalled

ok      github.com/Dynatrace/dynatrace-operator/test/scenarios/standard 153.500s

```

Resolves https://dt-rnd.atlassian.net/browse/K8S-11454

## How can this be tested?

Run e2e tests into different concoles / against different clusters, but reuse tenant:

```
make test/e2e/edgeconnect
make test/e2e/supportarchive
```

### CODE REVIEW

**Review by 0sewa0 — COMMENTED** [review]
wouldn't the name of the `edgeconnect` CR also conflict? as the `ProvisionerMode` will create the edgeconnect-entries on the tenant named after the CR

**Comment by andriisoldatenko** [comment]
> edgeconnect

Yes you are right, you can't create 2 EdgeConnects with similar names:

> Name
>test
>Name must be unique across configurations. Name 'test' already exists in EdgeConnect configuration 'test'.

let me fix it

File: test/features/edgeconnect/edgeconnect.go [hunk 1]
```diff
@@ -49,13 +48,16 @@ func NormalModeFeature(t *testing.T) features.Feature {
 
 	edgeConnectTenantConfig := &tenantConfig{}
 
-	builder.Assess("create EC configuration on the tenant", createTenantConfig(secretConfig, edgeConnectTenantConfig))
+	testECname := fmt.Sprintf("test-edgeconnect-normal-%d", rand.IntN(100))             //nolint
+	testHostPattern := fmt.Sprintf("e2eTestHostPattern%d.internal.org", rand.IntN(100)) //nolint
```
Comment (0sewa0) [inline test/features/edgeconnect/edgeconnect.go:52]: could we avoid putting the `//nolint` thing everywhere? why have linters if we just turn them off?

I know that they are "just tests" but still

also is rand of 100 random enough? we used to have similar logic for the network-zone test, and it was still flaky
- couldn't we use `github.com/google/uuid` ? or would that be overkill? (we already have it as a indirect dependency )

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go:52]: sure, i can try it.

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go]: @0sewa0 fixed in 131b746a

File: test/features/edgeconnect/edgeconnect.go [hunk 2]
```diff
@@ -14,6 +13,7 @@ import (
 	controller "github.com/Dynatrace/dynatrace-operator/pkg/controllers/edgeconnect"
 	"github.com/Dynatrace/dynatrace-operator/test/helpers"
 	"github.com/Dynatrace/dynatrace-operator/test/helpers/components/edgeconnect"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/rand"
```
Comment (0sewa0) [inline test/features/edgeconnect/edgeconnect.go:16]: this was used for the network-zone tests(and is actually dead code 😅 ), that were still flaky, is it the best idea to use it ? 😅 

what is the problem with the `uuid` package ?

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go:16]: ooops, one second.

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go:16]: but we don't use uuid anywhere...

Comment (0sewa0) [reply test/features/edgeconnect/edgeconnect.go:16]: so we implement/maintain an inferior inhouse thing for the same purpose while it's already part of our go.mod 😅

Comment (chrismuellner) [reply test/features/edgeconnect/edgeconnect.go:16]: let's use the uuid for a proper subdomain, instead of a single int with high chance of collision

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go]: fixed in e82d1097

also removed dead code.

File: test/features/edgeconnect/edgeconnect.go [hunk 3]
```diff
@@ -49,13 +49,17 @@ func NormalModeFeature(t *testing.T) features.Feature {
 
 	edgeConnectTenantConfig := &tenantConfig{}
 
-	builder.Assess("create EC configuration on the tenant", createTenantConfig(secretConfig, edgeConnectTenantConfig))
+	randSuffix := uuid.NewString()[:defaultRandomLength]
+	testECname := "test-edgeconnect-normal-" + randSuffix
+	testHostPattern := fmt.Sprintf("e2eTestHostPattern%s.internal.org", randSuffix)
+
```
Comment (0sewa0) [inline test/features/edgeconnect/edgeconnect.go:55]: why overcomplicate it with only using part of the uuid? also kinda defeats the purpose of the uuid 😅 

```
	testECname := uuid.NewString()
	testHostPattern := fmt.Sprintf("e2eTestHostPattern.%s.internal.org", testECname)
```

the name `test-edgeconnect-normal-` serves no real purpose, and you can perfectly see what was created by the e2e tests from the logs: (also the ec tests are so fast, that you will not see the deployment if you are not directly looking at it 😅 )
```
--- PASS: TestStandard (46.06s)
    --- PASS: TestStandard/edgeconnect-install (13.18s)
        --- PASS: TestStandard/edgeconnect-install/create_EC_configuration_on_the_tenant (1.07s)
        --- PASS: TestStandard/edgeconnect-install/create_client_secret (0.28s)
        --- PASS: TestStandard/edgeconnect-install/'0b038866-29ce-4c9e-b8eb-dc77791c6d18'_edgeconnect_created (0.30s)
        --- PASS: TestStandard/edgeconnect-install/'0b038866-29ce-4c9e-b8eb-dc77791c6d18'_edgeconnect_phase_changes_to_'Running' (5.15s)
        --- PASS: TestStandard/edgeconnect-install/check_EC_configuration_on_the_tenant (0.39s)
        --- PASS: TestStandard/edgeconnect-install/delete_EdgeConnect_CR (5.28s)
        --- PASS: TestStandard/edgeconnect-install/check_if_EC_configuration_is_deleted_on_the_tenant (0.28s)
    --- PASS: TestStandard/edgeconnect-install-provisioner (17.65s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/create_edgeconnect_client_secret (0.13s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/'8b312735-b051-42ff-98f4-18a693865d43'_edgeconnect_created (0.13s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/'8b312735-b051-42ff-98f4-18a693865d43'_edgeconnect_phase_changes_to_'Running' (5.16s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/get_tenant_config (0.25s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/get_EC_status (0.13s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_exists_on_the_tenant (0.27s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern (0.35s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/update_hostPatterns (5.28s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern2 (0.28s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/delete_EC_custom_resource (5.29s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_is_deleted_on_the_tenant (0.25s)
    --- PASS: TestStandard/edgeconnect-install-k8s-automation (13.04s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/create_ServiceAccount (0.14s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/create_edgeconnect_client_secret (0.13s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/'7ba14241-09b0-4d07-a8b7-cab65118d65c'_edgeconnect_created (0.13s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/'7ba14241-09b0-4d07-a8b7-cab65118d65c'_edgeconnect_phase_changes_to_'Running' (5.14s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/get_tenant_config (0.25s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/get_EC_status (0.13s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_exists_on_the_tenant (0.27s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_hostPatterns_-_k8s_automation (0.34s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_exists_on_the_tenant (0.39s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/delete_EC_custom_resource (5.28s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_is_deleted_on_the_tenant (0.25s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_is_deleted_on_the_tenant (0.30s)
PASS
out: helm uninstall dynatrace-operator \
			--namespace dynatrace
release "dynatrace-operator" uninstalled

ok  	github.com/Dynatrace/dynatrace-operator/test/scenarios/standard	129.859s
```

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go:55]: yeah it was my idea too, let's go with full uuid names. :)

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go]: fixed in 3af244ec

**Review by 0sewa0 — DISMISSED** [review]
```
--- PASS: TestStandard (136.53s)
    --- PASS: TestStandard/edgeconnect-install (103.78s)
        --- PASS: TestStandard/edgeconnect-install/create_EC_configuration_on_the_tenant (1.05s)
        --- PASS: TestStandard/edgeconnect-install/create_client_secret (0.25s)
        --- PASS: TestStandard/edgeconnect-install/'467a6a66-93ea-4bbc-a9c0-b9c7aa73a8ad'_edgeconnect_created (0.30s)
        --- PASS: TestStandard/edgeconnect-install/'467a6a66-93ea-4bbc-a9c0-b9c7aa73a8ad'_edgeconnect_phase_changes_to_'Running' (95.13s)
        --- PASS: TestStandard/edgeconnect-install/check_EC_configuration_on_the_tenant (0.93s)
        --- PASS: TestStandard/edgeconnect-install/delete_EdgeConnect_CR (5.29s)
        --- PASS: TestStandard/edgeconnect-install/check_if_EC_configuration_is_deleted_on_the_tenant (0.40s)
    --- PASS: TestStandard/edgeconnect-install-provisioner (17.62s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/create_edgeconnect_client_secret (0.16s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/'16d0fa3c-946f-4eeb-b16f-b10dc1b24f49'_edgeconnect_created (0.16s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/'16d0fa3c-946f-4eeb-b16f-b10dc1b24f49'_edgeconnect_phase_changes_to_'Running' (5.15s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/get_tenant_config (0.25s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/get_EC_status (0.13s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_exists_on_the_tenant (0.27s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern (0.29s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/update_hostPatterns (5.28s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_hostPatterns_on_the_tenant_-_testHostPattern2 (0.27s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/delete_EC_custom_resource (5.27s)
        --- PASS: TestStandard/edgeconnect-install-provisioner/check_if_EC_configuration_is_deleted_on_the_tenant (0.25s)
    --- PASS: TestStandard/edgeconnect-install-k8s-automation (12.96s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/create_ServiceAccount (0.15s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/create_edgeconnect_client_secret (0.13s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/'f6803f1d-a688-4ca1-a646-31f77f383227'_edgeconnect_created (0.15s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/'f6803f1d-a688-4ca1-a646-31f77f383227'_edgeconnect_phase_changes_to_'Running' (5.15s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/get_tenant_config (0.25s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/get_EC_status (0.13s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_exists_on_the_tenant (0.31s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_hostPatterns_-_k8s_automation (0.27s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_exists_on_the_tenant (0.30s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/delete_EC_custom_resource (5.28s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_EC_configuration_is_deleted_on_the_tenant (0.26s)
        --- PASS: TestStandard/edgeconnect-install-k8s-automation/check_if_settings_object_is_deleted_on_the_tenant (0.30s)
PASS
out: helm uninstall dynatrace-operator \
			--namespace dynatrace
release "dynatrace-operator" uninstalled

ok  	github.com/Dynatrace/dynatrace-operator/test/scenarios/standard	226.768s
```

File: test/features/edgeconnect/edgeconnect.go [hunk 4]
```diff
@@ -49,13 +48,16 @@ func NormalModeFeature(t *testing.T) features.Feature {
 
 	edgeConnectTenantConfig := &tenantConfig{}
 
-	builder.Assess("create EC configuration on the tenant", createTenantConfig(secretConfig, edgeConnectTenantConfig))
+	testECname := uuid.NewString()
+	testHostPattern := fmt.Sprintf("e2eTestHostPattern%s.internal.org", testECname)
```
Comment (chrismuellner) [inline test/features/edgeconnect/edgeconnect.go:52]:
```suggestion
	testHostPattern := fmt.Sprintf("%s.e2eTestHostPattern.internal.org", testECname)
```

Comment (andriisoldatenko) [reply test/features/edgeconnect/edgeconnect.go]: fixed in eb73a768

**Review by 0sewa0 — DISMISSED** [review]

**Review by chrismuellner — APPROVED** [review]

**Review by StefanHauth — APPROVED** [review]

**Review by 0sewa0 — APPROVED** [review]
both edgeconnect and supportarchive test past

**Review by waodim — APPROVED** [review]
ran both tests several times - worked for me 👍

**Pull request merged**